### PR TITLE
added check to disallow super.private variable access and test case added

### PR DESCRIPTION
**Pull request by vivek12345**
| Q                        | A 
| ------------------------ | ---
| Fixed Issues?            | Fixes https://github.com/babel/babel/issues/10470 
| Patch: Bug Fix?          | Yes
| Major: Breaking Change?  | No
| Minor: New Feature?      | No
| Tests Added + Pass?      | Yes
| Documentation PR Link    | 
| Any Dependency Changes?  | No
| License                  | MIT

### CODE REVIEW

File: packages/babel-parser/src/parser/expression.js [hunk 1]
```diff
@@ -1170,7 +1179,9 @@ export default class ExpressionParser extends LValParser {
     if (node.property.name !== propertyName || containsEsc) {
       this.raise(
         node.property.start,
-        `The only valid meta property for ${meta.name} is ${meta.name}.${propertyName}`,
+        `The only valid meta property for ${meta.name} is ${
```
Comment (vivek12345) [inline packages/babel-parser/src/parser/expression.js]: this is my linter doing something crazy. I will try and remove this change.

File: packages/babel-parser/src/parser/expression.js [hunk 2]
```diff
@@ -655,6 +655,15 @@ export default class ExpressionParser extends LValParser {
       node.object = base;
       node.property = this.parseMaybePrivateName();
       node.computed = false;
+      if (
+        node.property.type === "PrivateName" &&
+        node.object.type === "Super"
+      ) {
+        this.raise(
+          startPos,
+          "super is not allowed to be called on a private identifier of a class",
```
Comment (nicolo-ribaudo) [inline packages/babel-parser/src/parser/expression.js]: Well, in this case `super` is not being called (it's not `super(#x)`).
What about `Private fields can't be accessed on super?`

Comment (vivek12345) [reply packages/babel-parser/src/parser/expression.js]: Yes that message makes more sense. I will make the changes accordingly and also update the tests. Thank you

**Comment by vivek12345** [comment]
@nicolo-ribaudo when I am doing
```
make test-test262-update-whitelist
```
I get the following error:
```
Error: Unsupported version of Test262: '4.0.0'
```

Any help here?

**Comment by nicolo-ribaudo** [comment]
Could you try running `yarn upgrade test262-stream`?

**Comment by vivek12345** [comment]
That did the trick. Thank you 👍

**Pull request merged**